compose/foundation/foundation/src/test/kotlin/androidx/compose/foundation/text/selection/SelectionTest.kt

```diff
@@ -70,147 +70,118 @@ class SelectionTest {
 
     @Test
     fun selection_merge_handles_not_cross() {
-        val startOffset1 = 9
-        val endOffset1 = 20
-        val selectableKey1 = 1L
-        val startAnchor1 = Selection.AnchorInfo(
-            direction = ResolvedTextDirection.Ltr,
-            offset = startOffset1,
-            selectableId = selectableKey1
-        )
-        val endAnchor1 = Selection.AnchorInfo(
-            direction = ResolvedTextDirection.Ltr,
-            offset = endOffset1,
-            selectableId = selectableKey1
-        )
-        val selection1 = Selection(
-            start = startAnchor1,
-            end = endAnchor1,
-            handlesCrossed = false
-        )
-        val startOffset2 = 0
-        val endOffset2 = 30
-        val selectableKey2 = 2L
-        val startAnchor2 = Selection.AnchorInfo(
-            direction = ResolvedTextDirection.Ltr,
-            offset = startOffset2,
-            selectableId = selectableKey2
-        )
-        val endAnchor2 = Selection.AnchorInfo(
-            direction = ResolvedTextDirection.Ltr,
-            offset = endOffset2,
-            selectableId = selectableKey2
-        )
-        val selection2 = Selection(
-            start = startAnchor2,
-            end = endAnchor2,
-            handlesCrossed = false
-        )
-
+        val selection1 = makeSelection(9, 20, 1L, false)
```

Comment:
Just to be sure I don't miss some detail: the tests check the same logic as before the change? The change here is just a reflective?

Reply:
While the refactoring is absolutely a good thing to do, we need to consider the following nuances:
This code is in the common source set, so the changes may come from the upstream later and there would be one more place with possible conflict. To overcome this, we can upstream this PR (not urgent, just to decide the next steps for ourselves).

Reply:
> Just to be sure I don't miss some detail: the tests check the same logic as before the change? The change here is just a refactoring + new tests for the fix?

Yep